pages/dkp/kommander/2.2/dkp-upgrade/upgrade-konvoy/enterprise/index.md

```diff
@@ -0,0 +1,164 @@
+
+1. Run the following upgrade command for the CAPI components.
+
+```bash
+dkp upgrade capi-components
+```
+
+The command should output something similar to the following:
+
+```bash
+✓ Upgrading CAPI components
+✓ Waiting for CAPI components to be upgraded
+✓ Initializing new CAPI components
+✓ Deleting Outdated Global ClusterResourceSets
+```
+
+1. Once finished, verify that the components were upgraded using the `kubectl get provider` command.
+
+```bash
+kubectl get provider --all-namespaces
+NAMESPACE                           NAME                            AGE     TYPE                     PROVIDER         VERSION
+capa-system                         infrastructure-aws              5m54s   InfrastructureProvider   aws              v1.2.0-d2iq.0
+capi-kubeadm-bootstrap-system       bootstrap-kubeadm               6m2s    BootstrapProvider        kubeadm          v1.1.3-d2iq.3
+capi-kubeadm-control-plane-system   control-plane-kubeadm           5m58s   ControlPlaneProvider     kubeadm          v1.1.3-d2iq.3
+capi-system                         cluster-api                     6m5s    CoreProvider             cluster-api      v1.1.3-d2iq.3
+cappp-system                        infrastructure-preprovisioned   5m44s   InfrastructureProvider   preprovisioned   v0.5.0
+capv-system                         infrastructure-vsphere          57s     InfrastructureProvider   vsphere          v1.1.1
+capz-system                         infrastructure-azure            5m47s   InfrastructureProvider   azure                 v1.1.1
+```
+If the upgrade fails, review the prerequisites section and ensure that you've followed the steps in the [DKP upgrade overview][dkpup].
+
+## Upgrade the core addons
+
+To install the core addons, DKP relies on the `ClusterResourceSet` [Cluster API feature][CAPI]. In the CAPI component upgrade, we deleted the previous set of outdated global `ClusterResourceSets` because prior to DKP 2.2 some addons were installed using a global configuration. In order to support individual cluster upgrades, DKP 2.2 now installs all addons with a unique set of `ClusterResourceSet` and corresponding referenced resources, all named using the cluster’s name as a suffix. For example: `calico-cni-installation-my-aws-cluster`.
+
+<p class="message--warning"><strong>WARNING:</strong> If you have modified any of the <code>clusterResourceSet</code> definitions, these changes will <strong>not</strong> be preserved when running the command <code>dkp upgrade addons</code>. You can use the <code>--dry-run -o yaml</code> options to save the new configuration to a file and remake the same changes upon each upgrade.</p>
+
+Your cluster comes preconfigured with a few different core addons that provide functionality to your cluster upon creation. These include: CSI, CNI, Cluster Autoscaler, and Node Feature Discovery. New versions of DKP may come prebundled with newer versions of these addons. Perform the following steps to update these addons.
```

Comment:
Should we mention somewhere that this needs to run for each managed and management cluster with the `KUBECONFIG` configured for each cluster?

Reply:
Yes, i'll mention that in the Overview section beneath the stepwise instructions:

> Run all three steps on the management cluster (Kommander cluster) first. Then, run the second and third step on additional managed clusters (Konvoy clusters), one cluster at a time with the KUBECONFIG configured for each cluster.